modules/transaction-payment/src/lib.rs

```diff
@@ -1425,6 +1445,7 @@ where
 		payed: NegativeImbalanceOf<T>,
 	) -> Result<(), TransactionValidityError> {
 		let refund = Pallet::<T>::weight_to_fee(refund_weight);
+		let refund = Pallet::<T>::calculate_final_fee(refund, None);
```

Comment:
this is noop when multiplier is None. I think we want to pass refund amount instead of weight

Reply:
remove it now, leave to invoker to decide whether invoke `apply_multiplier_to_fee`/`calculate_final_fee` or not.